essencium-backend/src/main/java/de/frachtwerk/essencium/backend/service/AbstractUserService.java

```diff
@@ -258,6 +270,24 @@ public USER save(USER user) {
               }
             });
 
+    if (Objects.nonNull(SecurityContextHolder.getContext().getAuthentication())) {
+      var executingUser = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+      if (executingUser instanceof AbstractBaseUser<?>) {
+        boolean doModifyMyself = ((AbstractBaseUser<?>) executingUser).getId() == id;
+        boolean isAdmin =
+            ((AbstractBaseUser<?>) executingUser)
+                .getRoles()
+                .contains(roleService.getByName(RoleService.ADMIN));
+        if (doModifyMyself
+            && isAdmin
+            && !((Collection<Object>) fieldUpdates.get(USER_ROLE_ATTRIBUTE))
+                .contains(roleService.getByName(RoleService.ADMIN))) {
+          throw new NotAllowedException(
+              "You cannot remove the role 'ADMIN' from yourself. That is to ensure there's at least one ADMIN remaining.");
+        }
```

Comment:
Largely redundant code for lines 199 - 208. We should move this to a separate method and call it as required.